controller/tasks/etcd/ingestion_assignment.go

```diff
@@ -173,33 +173,57 @@ func (ia *ingestionAssignmentTask) Done() {
 	close(ia.stopChan)
 }
 
-func (ia *ingestionAssignmentTask) taskAssginmentLoop() {
-	namespaces, err := ia.namespaceMutator.ListNamespaces()
-	if err != nil {
-		ia.logger.Error(err)
-	}
-	for _, ns := range namespaces {
-		err = ia.watchManager.AddNamespace(ns)
+func (ia *ingestionAssignmentTask) startIngestionAssignment(hostName string) {
+	for {
+		select {
+		// waiting for new election status change
+		case <-ia.leaderElection.C():
+			if !ia.isLeader() {
+				continue
+			}
+		case <-ia.stopChan:
+			return
+		}
+
+		ia.logger.With("host", hostName).Infof("elected as leader")
+		namespaces, err := ia.namespaceMutator.ListNamespaces()
 		if err != nil {
-			ia.logger.Fatal(err)
+			ia.logger.Error(err)
+		}
+		for _, ns := range namespaces {
+			err = ia.watchManager.AddNamespace(ns)
+			if err != nil {
+				ia.logger.Fatal(err)
+			}
 		}
-	}
 
-	tickerChan := time.NewTicker(time.Duration(ia.intervalSeconds) * time.Second).C
-	for {
-		select {
-		// watch changes (i.e: heartbeat changes)
-		case ns := <-ia.watchManager.C():
-			if ia.isLeader() {
+		tickerChan := time.NewTicker(time.Duration(ia.intervalSeconds) * time.Second).C
+
+	loop:
+		for {
+			select {
+			case <- ia.leaderElection.C():
```

Comment:
do we need this? this process should already be leader past line 180, right?

Reply:
oh, is it for resign?

Reply:
yes, this is for when resign happens